### DISCUSSION TITLE
Multiple facets on same axis

### DISCUSSION THREAD

**Discussion by royfrancis** [observablehq/plot#2217] [Q&A] [answered]
My understanding is that when facetting, one variable can be provided along the x axis (`fx: "bla"`) and one along the y axis (`fy: "bla"`). But what if I have more variables to facet? Is it possible to have more than one variable on the same axis? ie; (`fx: ["bla1", "bla2"]`)

As an example, I would like to facet by color and clarity. I am mapping color to x facet and clarity to y facet. This is fine, but there are multiple y axes and I would like to have one y-axis to make comparison easier. Also, what if we have more than two variables to facet?

![Screenshot 2024-10-31 at 14 02 00](https://github.com/user-attachments/assets/23224c4e-9220-4119-8a1d-35177e1acac0)

<details>
<summary>Code</summary>

```js
filteredDiamonds = diamonds.filter(d => 
  (d.color === 'D' || d.color === 'E' || d.color === 'F') &&
  (d.clarity === 'I1' || d.clarity === 'IF' || d.clarity === 'SI1')
)

Plot.plot({
  title: "Boxplot",
  marks: [
    Plot.boxY(filteredDiamonds, {
      x: "cut",
      y: "depth",
      fx: "color",
      fy: "clarity"
    }),
    Plot.axisX({tickRotate: -45})
  ],
  grid: true
})
```
</details>

One solution is to create new variables by concatenating them and then using that to facet.

![Screenshot 2024-10-31 at 14 06 50](https://github.com/user-attachments/assets/6c8a26ad-26c6-4c6a-804b-564b05313ef6)

<details>
<summary>Code</summary>

filteredDiamonds1 = filteredDiamonds.map(d => ({
  ...d,
  clarity_color: `${d.clarity} ${d.color}`
}));

Plot.plot({
  title: "Boxplot",
  marks: [
    Plot.boxY(filteredDiamonds1, {
      x: "cut",
      y: "depth",
      fx: "clarity_color"
    }),
    Plot.axisX({tickRotate: -45})
  ],
  grid: true
})
</details>

This is not a great solution because it's messy and we lose the grouping. I am thinking of something like this where the grouping is maintained:

![Screenshot 2024-10-31 at 14 06 50 copy](https://github.com/user-attachments/assets/3b0525f7-b3ef-4a59-bb7b-ba868144cf08)

A working document for this issue: https://observablehq.com/d/5e28dce4416bb6c6

**Comment by Fil** [comment, marked as answer]
Here's an approach to nested facets:
https://observablehq.com/@observablehq/plot-nested-facets
